src/System.Private.ServiceModel/tools/test/SelfHostWcfService/WcfService.cs

```diff
@@ -367,6 +368,23 @@ public void ReturnContentType(string contentType)
             httpRespononseMessageProperty.Headers[HttpResponseHeader.ContentType] = contentType;
         }
 
+        public string EchoCookie()
+        {
+            string cookie = WebOperationContext.Current.IncomingRequest.Headers[HttpRequestHeader.Cookie];
+            return cookie;
+        }
+
+        public string EchoTimeAndSetCookie(string name)
```

Comment:
Can you put a comment either here, or in the calling test - perhaps something like: 
"Returns the current time and also sets the cookie 'name' to be the same time returned" 

This was a little confusing to read through as it wasn't apparent to me (from the calling side) that the method both returned the datetime as text and as a value set in the cookie.